Round robin_JB (#147)
jaybee84/ml-in-rd: content/03.heterogeneity.md

### Manage complex high-dimensional rare disease data

> In rare diseases, the high throughput ‘omic’ methods generate highly dimensional data - that is, data with many features (e.g., all of the mRNA transcripts in a sample), from a vanishingly small number of samples.

Reviewer: ```suggestion
In rare diseases, the high throughput ‘omic’ methods generate highly dimensional data – data with many features such as all of the mRNA transcripts in a sample – from a vanishingly small number of samples.
```

> This lack of samples gives rise to the “curse of dimensionality” (i.e., few samples but many features), which becomes an impediment in analyzing feature-rich data in sample-deficient contexts such as rare disease [@doi:10.1038/nrc2294] 

Reviewer: ```suggestion
A lack of samples gives rise to the “curse of dimensionality” (i.e., few samples but many features), which is an impediment in analyzing feature-rich data in sample-deficient contexts such as rare disease.[@doi:10.1038/nrc2294] 
```

In particular, increased numbers of features results in increased sparsity (missing observations), more dissimilarity between samples, and increased redundancy between individual features or combinations of features [@doi:10.1038/s41592-018-0019-x]; the consequence of this additional data is a challenging prediction problem. 
Furthermore, rare disease data collection and aggregation methods can add to these challenges by introducing technical variability into the data at hand.
In this section, we will discuss strategies like simplifying data and addressing technical artifacts through dimensionality reduction.

Dimensionality reduction methods like multidimensional scaling (MDS), principal components analysis (PCA), t-distributed stochastic neighbor embedding (t-SNE), and uniform manifold approximation and projection (UMAP) can help ‘compress’ information from a large number of features into a smaller number of features  in an unsupervised manner [@doi:10.1007/978-3-540-33037-0; @doi:10.1098/rsta.2015.0202, @https://lvdmaaten.github.io/publications/papers/JMLR_2008.pdf; @https://arxiv.org/abs/1802.03426] . 

> These methods not only help in reducing the number of features in various types of data [@doi:10.1016/j.media.2020.101660; @doi:10.1038/ncomms14825]; but can also be used to visualize structure or artifacts in the data (e.g. [@doi:10.1038/s41467-019-13056-x]), define subgroups within data (e.g. [@doi:10.1038/s41467-020-15351-4], or for feature selection/extraction during application of specific machine learning models.[@doi:10.1007/978-3-030-03243-2_299-1] (Figure {@fig:1})

Reviewer: ```suggestion
These methods not only help in reducing the number of features in various types of data [@doi:10.1016/j.media.2020.101660; @doi:10.1038/ncomms14825], but can also be used to visualize structure or artifacts in the data (e.g. [@doi:10.1038/s41467-019-13056-x]), define subgroups within data (e.g. [@doi:10.1038/s41467-020-15351-4], or for feature selection or extraction during application of specific machine learning models.[@doi:10.1007/978-3-030-03243-2_299-1] (Figure {@fig:1})
```

Dimensionality reduction, or more fundamentally representation learning, learns low-dimensional representations or composites of features from the raw data. 
For example, representation learning through matrix factorization can extract composite features from transcriptomics datasets that are made of combinations of gene expression levels found in the training data[@doi:10.1038/s41467-020-14666-6], and then use them to interpret test input data [@doi:10.1093/bioinformatics/btq503; @doi:10.1186/s13059-020-02021-3].
To ensure that the learned representations are generalizable, the features learned by the model can be constrained through methods like regularization [@doi:10.1371/journal.pgen.1004754, @doi:10.1002/sim.6782]. 
Representation learning generally requires many samples in complex biological systems, and thus may appear to aggravate the curse of dimensionality. 

> But it can be a powerful tool to learn low-dimensional patterns from large datasets and then find those patterns in smaller, related datasets. 

Reviewer: ```suggestion
However, it can be a powerful tool to learn low-dimensional patterns from large datasets and then find those patterns in smaller, related datasets. 
```
